src/rev.cpp.in

```diff
@@ -82,29 +83,22 @@ const char *freerct_userdata_prefix()
 {
 	if (!_freerct_userdata_prefix_cache.empty()) return _freerct_userdata_prefix_cache.c_str();
 
-	_freerct_userdata_prefix_cache = _freerct_userdata_prefix_base;
-
-	/* While the USERDATA_PREFIX is expanded at compile-time, home directory expansion has to be performed at runtime to allow shipping binaries. */
-	std::string substitute;
-	for (;;) {
-		const size_t pos = _freerct_userdata_prefix_cache.find('~');
-		if (pos == std::string::npos) break;
-
-		if (substitute.empty()) {
-			for (auto& var : {"HOME", "USERPROFILE", "HOMEPATH", "APPDATA"}) {
-				const char *environment_variable = getenv(var);
-				if (environment_variable != nullptr && environment_variable[0] != '\0') {
-					substitute = environment_variable;
-					break;
-				}
-			}
-			if (substitute.empty()) {
-				fprintf(stderr, "Unable to locate the user home directory. Set the HOME environment variable to fix the problem.\n");
-				NOT_REACHED();
-			}
+	if (!_freerct_userdata_prefix_override.empty()) {  // The user manually set a location.
+		_freerct_userdata_prefix_cache = _freerct_userdata_prefix_override;
+	} else {  // Try to follow the XDG basedir specification.
+		const char *xdg_data_home = getenv("XDG_DATA_HOME");
+		if (xdg_data_home != nullptr && xdg_data_home[0] != '\0') {
+			_freerct_userdata_prefix_cache = xdg_data_home;
+			_freerct_userdata_prefix_cache += "/freerct";
+		} else {  // The XDG spec is not applicable, fall back to defaults.
+			_freerct_userdata_prefix_cache = GetUserHomeDirectory();
+			/* Set a reasonable default path relative to the home directory. */
+#ifdef _WIN32
+			_freerct_userdata_prefix_cache += "\\.freerct";
+#else
+			_freerct_userdata_prefix_cache += "/.local/share/freerct";
+#endif
```

Comment:
The `// explanation` text after the `else` make this a bit of a wall of text.
I think it's easier to read if you insert a newline after the `{` as the `if` / `else` structure becomes more visible then.